### PR TITLE
[FVM] contracts.go refactor

### PR DESCRIPTION
**Pull request by janezpodhostnik**
ref: https://github.com/dapperlabs/flow-go/issues/6229

Some cleanup on `contracts.go` see referenced issue fro more details.

### CODE REVIEW

**Review by turbolent — APPROVED** [review]
Nice! Thank you

File: fvm/handler/contract.go [hunk 1]
```diff
@@ -148,15 +137,14 @@ func (l contractUpdateList) Less(i, j int) bool {
...
-
 	updatedKeys := h.UpdateKeys()
 	updateList := make(contractUpdateList, 0)
 
-	for _, uk := range h.draftUpdates {
+	for k, uk := range h.draftUpdates {
 		updateList = append(updateList, uk)
+
+		// delete as we go to clear h.draftUpdates
+		delete(h.draftUpdates, k)
```
[the remark this answers is not in the document]
Comment (ramtinms) [reply fvm/handler/contract.go:147]: also is it safe you remove elements while iterating ?

**Review by ramtinms — APPROVED** [review]
Just a minor comment

File: fvm/handler/contract.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (janezpodhostnik) [reply fvm/handler/contract.go:147]: I just double checked. The answer is:

<details>
    <summary>Yes</summary>
      
    func createRandomBytes(l uint64) []byte {
	      b := make([]byte, l)
	      rand.Read(b)
	      return b
      }
      
      func createRandomShortString() string {
	      l := 10
	      return string(createRandomBytes(uint64(l)))
      }
      
      func createRandomLongString() string {
	      l := 1000
	      return string(createRandomBytes(uint64(l)))
      }
      
      func Benchmark_NewMap(b *testing.B) {
	      b.ReportAllocs()
	      m := map[string]string{}
	      n := 100
	      for i := 0; i < b.N; i++ {
		      // fill map
		      for j := 0; j < n; j++ {
			      m[createRandomShortString()] = createRandomLongString()
		      }
		      
		      // iterate through the map and do "something with the values
		      sum := 0
		      for k, v := range m {
			      sum += len(v) + len(k)
		      }
      
		      // clear the map
		      m = map[string]string{}
	      }
      }
      
      func Benchmark_DeleteFromMap(b *testing.B) {
	      b.ReportAllocs()
	      m := map[string]string{}
	      n := 100
	      for i := 0; i < b.N; i++ {
		      // fill map
		      for j := 0; j < n; j++ {
			      m[createRandomShortString()] = createRandomLongString()
		      }
      
		      // iterate through the map and do "something with the values and delete them
		      sum := 0
		      for k, v := range m {
			      sum += len(v) + len(k)
			      delete(m, k)
		      }
	      }
      }
    /*
      goos: darwin
      goarch: amd64
      pkg: awesomeProject1
      cpu: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
      Benchmark_NewMap
      Benchmark_NewMap-8          	  453532	    181641 ns/op	  218507 B/op	     411 allocs/op
      Benchmark_DeleteFromMap
      Benchmark_DeleteFromMap-8   	  500053	    148677 ns/op	  208538 B/op	     400 allocs/op
      PASS
    */
</details>

and [yes](https://stackoverflow.com/a/23230406/9815813)

**Review by m4ksio — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**